real_time/surface_src/java_src/Alice/src/test/java/com/roboclub/robobuggy/nodes/sensors/FullLoggingTest.java

```diff
@@ -0,0 +1,206 @@
+
+import com.roboclub.robobuggy.messages.BrakeStateMessage;
+import com.roboclub.robobuggy.messages.EncoderMeasurement;
+import com.roboclub.robobuggy.messages.GpsMeasurement;
+import com.roboclub.robobuggy.messages.GuiLoggingButtonMessage;
+import com.roboclub.robobuggy.messages.ImageMessage;
+import com.roboclub.robobuggy.messages.MagneticMeasurement;
+import com.roboclub.robobuggy.messages.SteeringMeasurement;
+import com.roboclub.robobuggy.ros.Message;
+import com.roboclub.robobuggy.ros.MessageListener;
+import com.roboclub.robobuggy.ros.NodeChannel;
+import com.roboclub.robobuggy.ros.Publisher;
+import com.roboclub.robobuggy.ros.Subscriber;
+import com.roboclub.robobuggy.simulation.SensorPlayer;
+import org.junit.Before;
+import org.junit.Test;
+
+import javax.imageio.ImageIO;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Random;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+/**
+ * A class that checks to see if we can log and play back correctly
+ */
+public class FullLoggingTest {
+
+    private Publisher gpsPub;
+    private Publisher imuPub;
+    private Publisher encoderPub;
+    private Publisher steeringPub;
+    private Publisher brakePub;
+    private Publisher videoPub;
+
+    private ArrayList<Message> masterList;
+    private ArrayList<Message> simList;
+
+    private boolean isPlaybackOccuring;
+
+    private static final int LOG_FILE_LENGTH = 100;
+    private static final int TIME_DELAY = 100;
+    private static final boolean USE_DELAYS = true;
+
+    /**
+     * initializes all of the subscribers and publishers,
+     */
+    @Before
+    public void oneTimeSetup() {
+
+        gpsPub = new Publisher(NodeChannel.GPS.getMsgPath());
+        imuPub = new Publisher(NodeChannel.IMU_MAGNETIC.getMsgPath());
+        encoderPub = new Publisher(NodeChannel.ENCODER.getMsgPath());
+        steeringPub = new Publisher(NodeChannel.STEERING.getMsgPath());
+        brakePub = new Publisher(NodeChannel.BRAKE_STATE.getMsgPath());
+        videoPub = new Publisher(NodeChannel.PUSHBAR_CAMERA.getMsgPath());
+
+        MessageListener addToSimList = (topicName, m) -> {
+            if (isPlaybackOccuring) {
+                simList.add(m);
+            }
+        };
+
+        new Subscriber("fullLoggingTest", NodeChannel.GPS.getMsgPath(), addToSimList);
+        new Subscriber("fullLoggingTest", NodeChannel.IMU_MAGNETIC.getMsgPath(), addToSimList);
+        new Subscriber("fullLoggingTest", NodeChannel.ENCODER.getMsgPath(), addToSimList);
+        new Subscriber("fullLoggingTest", NodeChannel.STEERING.getMsgPath(), addToSimList);
+        new Subscriber("fullLoggingTest", NodeChannel.BRAKE_STATE.getMsgPath(), addToSimList);
+        new Subscriber("fullLoggingTest", NodeChannel.PUSHBAR_CAMERA.getMsgPath(), addToSimList);
+
+    }
+
+    /**
+     * Cleans out the master and check list
+     */
+    @Before
+    public void setup() {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [You may have misspelled a JUnit framework method (setUp or tearDown)](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2381518032/issues/source?bid=2938665&fileBranchId=3199438#l81)